test/micro/org/openjdk/bench/java/util/zip/InflaterInputStreams.java

```diff
@@ -0,0 +1,112 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.java.util.zip;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.Inflater;
+import java.util.zip.InflaterInputStream;
+import java.util.zip.DeflaterOutputStream;
+
+/**
+ * A simple benchmark to measure the performance improvements achieved by avoiding
+ * unnecessary native calls in InflaterInputStream::read() (see JDK-8281962).
+ *
+ * before JDK-8281962
+ * ------------------
+ * Benchmark                                     (size)  Mode  Cnt  Score   Error  Units
+ * InflaterInputStreams.inflaterInputStreamRead     256  avgt    5  2.571 ± 0.120  us/op
+ * InflaterInputStreams.inflaterInputStreamRead     512  avgt    5  2.861 ± 0.064  us/op
+ * InflaterInputStreams.inflaterInputStreamRead    4096  avgt    5  5.110 ± 0.278  us/op
+ *
+ * after JDK-8281962
+ * -----------------
+ * Benchmark                                     (size)  Mode  Cnt  Score   Error  Units
+ * InflaterInputStreams.inflaterInputStreamRead     256  avgt    5  2.332 ± 0.081  us/op
+ * InflaterInputStreams.inflaterInputStreamRead     512  avgt    5  2.691 ± 0.293  us/op
+ * InflaterInputStreams.inflaterInputStreamRead    4096  avgt    5  4.812 ± 1.038  us/op
+ *
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Thread)
+@Warmup(iterations = 3, time = 5, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 5, timeUnit = TimeUnit.SECONDS)
+public class InflaterInputStreams {
+
+    @Param({"256", "512", "4096"})
+    private int size;
+    private byte[] chars;
+    private byte[] words;
+    private static final int MAX_SIZE = 5000; // Should be bigger than the biggest size @Param
+    private static byte[] inflated = new byte[MAX_SIZE];
```

Comment:
If it isn't important for the benchmark, I'd remove the hard-coded `MAX_SIZE` and make `inflated` non-static and allocate it in `beforeRun` using `size`. (Future enhancements could very well be interested in assessing the performance of larger sized entries, and we shouldn't put up trip-wires)

Reply:
Sure , you're absolutely right. Updated as suggested.

I've also removed some non-ASCII characters from the comment to avoid problems with javac.

Finally, a year isn't required in the Amazon copyright header. We specially removed all years a while back.

PS: and thanks a lot for your review :)